src/charm.py

```diff
@@ -73,10 +186,64 @@ def _install_apt_packages(self, packages: list):
         except apt.PackageNotFoundError:
             logger.error("a specified package not found in package cache or on system")
             self.unit.status = BlockedStatus("Failed to install packages")
-        except apt.PackageError as e:
+        except Exception as e:
             logger.error("could not install package. Reason: %s", e.message)
             self.unit.status = BlockedStatus("Failed to install packages")
 
+    @property
+    def unit_ips(self) -> list:
+        """Retrieve IP addressses associated with mongoDB application
+        Returns:
+            list (str): IP address associated with mongoDB application
+        """
+        peer_addresses = [
+            str(self.peers.data[unit].get("private_address")) for unit in self.peers.units
+        ]
+
+        self_address = str(self.model.get_binding(PEER).network.bind_address)
+        logger.debug("this machines address %s", self_address)
+        addresses = []
+        if peer_addresses:
+            addresses.extend(peer_addresses)
+        addresses.append(self_address)
+        return addresses
+
+    @property
+    def config(self) -> dict:
+        """Retrieve config options for mongo
+        Returns:
+            dict: Config options for mogno
+        """
```

Comment:
```suggestion
    def config(self) -> Dict:
        """Retrieve config options for MongoDB.
        
        Returns:
            A dictionary representation of MongoDB config options
        """
```

Reply:
One broader point, throughout you refer to MongoDB as

- mongodb
- mongoDB
- MongoDB
- mongo

Probably better to pick one :)

Reply:
Good idea, picked MongoDB and hopefully caught all of them 😅